### PR TITLE
Separate network IDs from save IDs

### PR DESCRIPTION
**Pull request by Xd-pro**
Allows multiple entity types to have the same network ID.

### CODE REVIEW

File: server/world/entity.go [hunk 1]
```diff
@@ -41,6 +42,12 @@ type TickerEntity interface {
 	Tick(w *World, current int64)
 }
 
+type NetworkedEntity interface {
```
Comment (DaPigGuy) [inline server/world/entity.go:45]: Missing a doc here

File: server/world/entity.go [hunk 2]
```diff
@@ -41,6 +42,12 @@ type TickerEntity interface {
...
+	Entity
+	// Returns the network type ID of the entity. This is NOT the save ID.
```
Comment (DaPigGuy) [inline server/world/entity.go:47]: This doc should begin with `// NetworkTypeID`

File: server/world/entity.go [hunk 3]
```diff
@@ -41,6 +42,13 @@ type TickerEntity interface {
 	Tick(w *World, current int64)
 }
 
+// NetworkEncodeableEntity is an Entity where the save ID and network ID is not the same.
```
Comment (DaPigGuy) [inline server/world/entity.go:45]:
```suggestion
// NetworkEncodeableEntity is an Entity where the save ID and network ID are not the same.
```

File: server/world/entity.go [hunk 4]
```diff
@@ -41,6 +42,13 @@ type TickerEntity interface {
...
+type NetworkEncodeableEntity interface {
+	Entity
```
Comment (Sandertv) [inline server/world/entity.go:47]: No reason to have this embed `Entity`.

File: server/entity/text.go [hunk 5]
```diff
@@ -24,11 +24,16 @@ func (t *Text) Name() string {
...
 }
 
-// EncodeEntity returns the ID for falling blocks.
+// EncodeEntity returns the ID for text.
 func (t *Text) EncodeEntity() string {
 	return "dragonfly:text"
 }
 
+// NetworkTypeID returns the network type ID for falling blocks.
+func (*Text) NetworkTypeID() string {
```
Comment (Sandertv) [inline server/entity/text.go:33]: This method should be called `NetworkEncodeEntity`.

File: server/world/entity.go [hunk 6]
```diff
@@ -41,6 +42,13 @@ type TickerEntity interface {
 	Tick(w *World, current int64)
 }
 
+// NetworkEncodeableEntity is an Entity where the save ID and network ID is not the same.
+type NetworkEncodeableEntity interface {
```
Comment (Sandertv) [inline server/world/entity.go:46]: I'd consider moving this to the session package, as that's the only place where the interface is used.

**Review by Sandertv — CHANGES_REQUESTED** [review]
I've left a couple of comments. I think this is ready to go once those and the ones left by @DaPigGuy are resolved.

File: server/world/entity.go [hunk 7]
```diff
@@ -41,6 +42,8 @@ type TickerEntity interface {
 	Tick(w *World, current int64)
 }
 
+
+
```
Comment (DaPigGuy) [inline server/world/entity.go:46]: Please revert all changes to this file

File: server/entity/text.go (hunk 5 above)
Comment (DaPigGuy) [inline server/entity/text.go:33]:
```suggestion
func (*Text) NetworkEncodeEntity() string {
```

**Review by Sandertv — APPROVED** [review]
This looks good. Thank you for the PR!

**Pull request merged**